ideascube/configuration/__init__.py

```diff
@@ -0,0 +1,87 @@
+import logging
+
+
+__all__ = [
+    'get_config',
+    'set_config',
+]
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_config(namespace, key):
+    """Get a configuration option
+
+    Args:
+        namespace (str): The configuration namespace.
+        key (str): The configuration key inside the namespace.
+
+    Returns:
+        The current value (str) for this option.
+
+    Raises:
+        ideascube.configuration.exceptions.NoSuchConfigurationNamespaceError:
+            The requested namespace does not exist.
+        ideascube.configuration.exceptions.NoSuchConfigurationKeyError:
+            The requested key does not exist in the requested namespace.
+    """
+    from .models import Configuration
+    from .registry import get_default_value, get_expected_type
+
+    default_value = get_default_value(namespace, key)
+    expected_type = get_expected_type(namespace, key)
+
+    try:
+        config = Configuration.objects.get(namespace=namespace, key=key)
+
+    except Configuration.DoesNotExist:
+        return default_value
+
+    if not isinstance(config.value, expected_type):
+        logger.error(
+            'The stored value for %s is of type %s instead of %s. This should'
+            ' never have happened.'
+            % (config, type(config.value), expected_type))
+        return default_value
+
+    return config.value
+
+
+def set_config(namespace, key, value, actor):
+    """Set a configuration option
+
+    Args:
+        namespace (str): The configuration namespace.
+        key (str): The configuration key inside the namespace.
+        value: The new value for the option. It should be of the type defined
+            in the registry for this configuration option.
+        actor (User): The staff user performing the configuration change, for
+            audit purposes.
+
+    Raises:
+        ideascube.configuration.exceptions.NoSuchConfigurationNamespaceError:
+            The requested namespace does not exist.
+        ideascube.configuration.exceptions.NoSuchConfigurationKeyError:
+            The requested key does not exist in the requested namespace.
+        ideascube.configuration.exceptions.InvalidConfigurationValueError:
+            The provided value is not the expected type.
+
+    Examples:
+        In the context of a view, with a `request` available, one could do:
+
+        >>> set_config('foo', 'bar', 'the value', request.user)
+    """
+    from .exceptions import InvalidConfigurationValueError
+    from .models import Configuration
+    from .registry import get_expected_type
+
+    expected_type = get_expected_type(namespace, key)
+
+    if not isinstance(value, expected_type):
+        raise InvalidConfigurationValueError(
+            namespace, key, expected_type, type(value))
+
+    Configuration.objects.update_or_create(
+        namespace=namespace, key=key,
+        defaults={'value': value, 'actor': actor})
```

Comment:
Do we update the existing object ?
(Then we need to update the date)

Or we create a new one with a different date ?
(And get the more recent one)